fast_llm/engine/training/config.py

```diff
@@ -364,9 +375,24 @@ class TrainerConfig(PretrainedFastLLMModelConfig, ExperimentConfig):
         desc="Configuration for the training optimizer and learning rate schedule.",
         hint=FieldHint.core,
     )
+    reference_models: dict[str, PretrainedFastLLMModelConfig] = Field(
+        default_factory=dict,
+        desc="Auxiliary models used during training, ex. for knowledge distillation.",
+        hint=FieldHint.feature,
+    )
 
     def _validate(self) -> None:
         self.training.export.setup(self.model)
+        self.model.validate()
+        if self.reference_models:
+            # TODO: Add support.
+            Assert.eq(self.model.distributed.pipeline_parallel, 1)
+            # TODO: Check if these work.
+            Assert.eq(self.model.distributed.tensor_parallel, 1)
+            Assert.eq(self.model.distributed.sequence_data_parallel, 1)
```

Comment:
It would be good to lift these restrictions because the reference models can and will be large, and the sequences will be long.

Reply:
is the distributed config shared between the student and the teacher models? if so, are we currently unable to use reference modes when training a model with pp, tp, sdp > 1?

Reply:
PP would be really hard, but TP and SDP won't be that hard. The tricky part will be to make loss functions that use it compatible.